[32mINFO    [0m pytest_operator.plugin:plugin.py:734 Connecting to existing model microk8s-localhost:test on unspecified cloud
[32mINFO    [0m integration.test_charm:test_charm.py:48 Deploying mysql, mysqlrouter and application
[32mINFO    [0m juju.model:model.py:2098 Deploying ch:amd64/jammy/mysql-test-app-63
[32mINFO    [0m juju.model:model.py:2098 Deploying ch:amd64/jammy/mysql-k8s-208
[32mINFO    [0m juju.model:model.py:2098 Deploying local:mysql-router-k8s-0
[32mINFO    [0m integration.test_charm:test_charm.py:79 Waiting for mysqlrouter to be in BlockedStatus
[32mINFO    [0m integration.test_charm:test_charm.py:85 Relating mysql, mysqlrouter and application
[33mWARNING [0m juju.model:model.py:1564 relate is deprecated and will be removed. Use integrate instead.
[33mWARNING [0m juju.model:model.py:1564 relate is deprecated and will be removed. Use integrate instead.
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  mysql-router-k8s/0 [executing] waiting: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  mysql-router-k8s/0 [idle] waiting: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  mysql-router-k8s/0 [idle] waiting: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  mysql-router-k8s/0 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  mysql-k8s/0 [executing] active: 
  mysql-k8s/1 [executing] active: Primary
  mysql-k8s/2 [idle] maintenance: joining the cluster
  mysql-router-k8s/0 [idle] active: 
  mysql-test-app/0 [idle] active: Last written value=439
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  mysql-k8s/0 [idle] active: 
  mysql-k8s/1 [idle] active: Primary
  mysql-k8s/2 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  mysql-router-k8s (waiting for exactly 2 units, current : 1)
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  mysql-router-k8s/0 [idle] active: 
  mysql-router-k8s/1 [executing] waiting: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  mysql-router-k8s/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  mysql-test-app (waiting for exactly 2 units, current : 1)
[32mINFO    [0m pytest_operator.plugin:plugin.py:862 Model status:

Model  Controller          Cloud/Region        Version  SLA          Timestamp
test   microk8s-localhost  microk8s/localhost  3.5.4    unsupported  01:44:11Z

App               Version                  Status  Scale  Charm             Channel      Rev  Address         Exposed  Message
mysql-k8s         8.0.37-0ubuntu0.22.04.3  active      3  mysql-k8s         8.0/edge     208  10.152.183.186  no       
mysql-router-k8s  8.0.37-0ubuntu0.22.04.3  active      2  mysql-router-k8s                 0  10.152.183.222  no       
mysql-test-app    0.0.2                    active      2  mysql-test-app    latest/edge   63  10.152.183.175  no       

Unit                 Workload  Agent  Address     Ports  Message
mysql-k8s/0          active    idle   10.1.18.14         
mysql-k8s/1*         active    idle   10.1.18.15         Primary
mysql-k8s/2          active    idle   10.1.18.13         
mysql-router-k8s/0*  active    idle   10.1.18.9          
mysql-router-k8s/1   active    idle   10.1.18.16         
mysql-test-app/0*    active    idle   10.1.18.8          
mysql-test-app/1     active    idle   10.1.18.17         

[32mINFO    [0m pytest_operator.plugin:plugin.py:868 Juju error logs:

unit-mysql-k8s-1: 01:41:11 ERROR unit.mysql-k8s/1.juju-log database:7: Failed to get cluster status for cluster-c9b47955aaf6871b6f43255ddb5c5360
unit-mysql-k8s-1: 01:41:11 ERROR unit.mysql-k8s/1.juju-log database:7: Failed to get cluster endpoints
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-mysql-k8s-1/charm/src/mysql_k8s_helpers.py", line 792, in update_endpoints
    rw_endpoints, ro_endpoints, offline = self.get_cluster_endpoints(get_ips=False)
  File "/var/lib/juju/agents/unit-mysql-k8s-1/charm/lib/charms/tempo_coordinator_k8s/v0/charm_tracing.py", line 734, in wrapped_function
    return callable(*args, **kwargs)  # type: ignore
  File "/var/lib/juju/agents/unit-mysql-k8s-1/charm/lib/charms/mysql/v0/mysql.py", line 1954, in get_cluster_endpoints
    raise MySQLGetClusterEndpointsError("Failed to get endpoints from cluster status")
charms.mysql.v0.mysql.MySQLGetClusterEndpointsError: Failed to get endpoints from cluster status

[32mINFO    [0m pytest_operator.plugin:plugin.py:947 Forgetting model main...